### ISSUE TITLE
chore: re-organize modules 

### ISSUE THREAD

**Issue by jeffbcross** [angular/angularfire2#253]
The current module and directory structure is based on early design of the project, and hasn't evolved with the project. The project should be better organized to reflect the organization of the library.

A quick proposal for discussion:

```
src/
 |- angularfire2.ts
 |- index.ts (main file, re-exporting only)
 |- tokens.ts
 |- interfaces.ts
 |- test-config.ts
 |- database/
   |- database.ts
   |- firebase_list_factory.ts
   |- firebase_list_factory.spec.ts
   |- firebase_list_observable.ts
   |- firebase_list_observable.spec.ts
   |- firebase_object_factory.ts
   |- firebase_object_factory.spec.ts
   |- firebase_object_observable.ts
   |- firebase_object_observable.spec.ts
   |- index.ts (just re-exporting, no logic)
   |- query_observable.ts
   |- query_observable.spec.ts
 |- auth/
   |- auth.ts
   |- auth.spec.ts
   |- auth_backend.ts
   |- auth_backend.spec.ts
   |- firebase_sdk_auth_backend.ts
   |- index.ts (just re-exporting, no logic)
 |- utils.ts
 |- worker/ (bindings for web workers)
   |- app.ts (renamed from angularfire2_worker_app.ts)
   |- render.ts (renamed from angularfire2_worker_render.ts)
 |- universal/ (bindings for universal, doesn't yet exist)
```

**Comment by jeffbcross** [comment]
CC @davideast

**Comment by davideast** [comment]
Love it. Exactly what I was thinking

**Comment by jeffbcross** [comment]
Fixed by 1a1ceb9
